openhcl/virt_mshv_vtl/src/processor/mod.rs

```diff
@@ -258,6 +259,12 @@ pub trait Backing: BackingPrivate {}
 
 impl<T: BackingPrivate> Backing for T {}
 
+/// Registers needed for supporting hypercall handling of gva to gpa translation
+pub struct HvTranslateGvaRegisters {
```

Comment:
This isn't my favorite, but changing TranslationRegisters itself to add the pat looked invasive to me as that's not the only structure that would have to change, and then I didn't want to go to the runner since it seemed unnecessary to make an ioctl when for SNP we can go to the vmsa. Open to thoughts.

Reply:
I think this is mostly fine. But we do want to avoid the extra vmcs read in the TDX ordinary emulation path (as opposed to the translate GVA hypercall path).

I think the cleanest fix is to avoid having this new struct and instead just have a method on the `HardwareIsolatedBacking` trait to get the PAT in the vendor specific way.

Alternatively, you could make TDX PAT querying faster by caching its value on MSR write. We already trap on PAT writes, so it would be straightforward to store the last PAT write in a cached value (and also update `set_access_control` to update the cache).

Or you could do both and save a tdcall in the hypercall path, too.